### ISSUE TITLE
Fix examples using temporary files

### ISSUE THREAD

**Issue by IndrajeetPatil** [jimhester/lintr#1887]
Ref: https://lintr.r-lib.org/dev/reference/trailing_blank_lines_linter.html

<img width="795" alt="Screenshot 2023-01-04 at 12 34 06" src="https://user-images.githubusercontent.com/11330453/210546636-1b5b413c-1950-428a-9a6a-41e3c040784a.png">

**Comment by IndrajeetPatil** [comment]
Hmm, I tried a few things, but nothing seems to work as expected, so I am out of ideas.

Any tips, @MichaelChirico, @AshesITR?

**Comment by MichaelChirico** [comment]
could be a permissions issue, check that file.create() actually succeeded

**Comment by IndrajeetPatil** [comment]
Yeah, you are right. The file is actually never created, and so there is nothing to link. 

I also tried the following:

```r
#' # will produce lints
#' file_path <- "foo.R"
#' writeLines("x <- 1\n", file_path)
#' readLines(file_path)
#' lint(
#'   file_path,
#'   linters = trailing_blank_lines_linter()
#' )
#' unlink(file_path)
```

But linting still doesn't work:

<img width="466" alt="Screenshot 2023-09-14 at 17 20 14" src="https://github.com/r-lib/lintr/assets/11330453/d902bccf-8477-411a-8720-7eb40d409307">

---

Should we just skip evaluating these examples (i.e. set `#' @examplesIf FALSE`)? Or is there any way around this problem?

**Comment by MichaelChirico** [comment]
instead of readLines, what's readChar(file_path, file.size(file_path))?

**Comment by IndrajeetPatil** [comment]
> instead of readLines, what's readChar(file_path, file.size(file_path))?

```r
readChar(file_path, file.size(file_path))
#> [1] "x <- 1\n\n"
```

**Comment by IndrajeetPatil** [comment]
The same issue is also seen in [`linters_with_defaults` docs](https://lintr.r-lib.org/reference/linters_with_defaults.html#ref-examples):

<img width="812" alt="Screenshot 2023-09-19 at 11 35 23" src="https://github.com/r-lib/lintr/assets/11330453/f9806baf-af14-4906-8e66-d6e2a0017840">

**Comment by IndrajeetPatil** [comment]
Reported in https://github.com/r-lib/lintr/issues/2243 The call to `lint()` fails in the ["Examples" section](https://lintr.r-lib.org/dev/reference/lint.html#ref-examples) of `lint()`, both in stable and devel versions of the website:

![image](https://github.com/r-lib/lintr/assets/52219252/1a1d88fc-9c8b-4f8f-be7c-a4a455a22317)

**Comment by MichaelChirico** [comment]
I think this is obsolete, I don't see those errors on the r-lib page anymore. Not sure what changed.
